tests/PHPStan/Rules/Methods/CallMethodsRuleTest.php

```diff
@@ -2628,4 +2628,33 @@ public function testBug7600(): void
 		$this->analyse([__DIR__ . '/data/bug-7600.php'], []);
 	}
 
+	public function testBug8058(): void
+	{
+		if (PHP_VERSION_ID < 80200) {
+			$this->markTestSkipped('Test requires PHP 8.2');
+		}
+		$this->checkThisOnly = false;
+		$this->checkNullables = false;
+		$this->checkUnionTypes = false;
+		$this->checkExplicitMixed = false;
+		$this->analyse([__DIR__ . '/data/bug-8058.php'], []);
+	}
+
+	public function testBug8058b(): void
+	{
+		if (PHP_VERSION_ID > 80200) {
+			$this->markTestSkipped('Test requires PHP before 8.2');
+		}
+		$this->checkThisOnly = false;
+		$this->checkNullables = false;
+		$this->checkUnionTypes = false;
+		$this->checkExplicitMixed = false;
+		$this->analyse([__DIR__ . '/data/bug-8058.php'], [
+			[
+				'Call to an undefined method mysqli::execute_query().',
```

Comment:
pretty interessting that only the method is undefined, but not the `\mysqli_execute_query` function

Reply:
There's a separate rule for that - CallToNonExistentFunctionRule.